lib/core/state/mongo-users-state.js

```diff
@@ -0,0 +1,46 @@
+const UserCommands = require('../commands/user-commands')
+const MongoClient = require('../clients/mongodb-client')
+const logger = require('../../helpers/logger')
+
+class MongoUsersState {
+  /**
+   *
+   * @param {Object} settings Settings
+   * @param {String} settings.users users to create
+   */
+  constructor(settings) {
+    const { connection } = settings
+    this.users = settings.users
+    // Client
+    this.client = settings.client || new MongoClient(connection.uri)
+  }
+
+  async apply() {
+    // this should sync the users
+    // for now it only creates/updates new ones
+    const userCommands = new UserCommands(this.client)
+
+    for (let i = 0; i < this.users.length; i += 1) {
```

Comment:
You can use `for of` here for more readability https://github.com/unumotors/mongo-hydra/blob/master/lib/core/state/replication.js#L58

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

(not a blocker)

Reply:
Done, I always get confused with the `for in` that can only be used with objects, thanks for pointing this out!